test/cbmc/proofs/dataHandlerCleanup/dataHandlerCleanup_harness.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * AWS IoT Over-the-air Update v3.1.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file dataHandlerCleanup_harness.c
+ * @brief Implements the proof harness for dataHandlerCleanup function.
+ */
+/* Include headers for ota agent. */
+#include "ota.h"
+#include "stubs.h"
+
+extern OtaAgentContext_t otaAgent;
+extern void dataHandlerCleanup( void );
+
+void dataHandlerCleanup_harness()
+{
+    OtaInterfaces_t otaInterface;
+
+    otaInterface.os.timer.stop = stopTimerStub;
+    otaInterface.os.event.send = sendEventStub;
+
+    /* Havoc otaAgent to non-deterministically set all the bytes in
+     * the structure. */
+    __CPROVER_havoc_object( &otaAgent );
+
+    /* otaAgent.pOtaInterface can never be NULL as it is always checked at the start of the OTA
+     * Agent specifically in receiveAndProcessOTAEvent function.*/
+    otaAgent.pOtaInterface = &otaInterface;
+
+    dataHandlerCleanup();
```

Comment:
Should this check that the job name has been cleared?

Reply:
No, we don't need to check if the job name has been cleared. In the CBMC proofs, we are trying to check if our code has any memory unsafe statements. If the jobName char string has been cleared or not will be checked by the unit-tests.